### PR TITLE
Add Elasticsearch client

### PR DESCRIPTION
**Pull request by wangch079**
## Part of https://github.com/elastic/enterprise-search-team/issues/2126

Add a Elasticsearch client (using [elasticsearch-ruby](https://github.com/elastic/elasticsearch-ruby))

#### Pre-Review Checklist
- [ ] Covered the changes with automated tests
- [x] Tested the changes locally
- [ ] ~Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)~ (Removed from template in this PR)

### CODE REVIEW

**Comment by wangch079** [comment]
>- [ ] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

This should be removed from the template? @elastic/ingestion-team

File: lib/connectors_shared/elasticsearch_client.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+      def connection_configs
+        es_config = ConnectorsApp::Config['elasticsearch']
+        configs = { :api_key => es_config['api_key'] }
+        if es_config['cloud_id']
+          configs[:cloud_id] = es_config['cloud_id']
+        elsif es_config['hosts']
+          configs[:hosts] = es_config['hosts']
+        else
+          raise 'Either elasitcsearch.cloud_id or elasticsearch.hosts should be configured.'
```
Comment (wangch079) [inline lib/connectors_shared/elasticsearch_client.rb:33]: This should be validated by config schema. I need to figure out how to make the rule that at least one of the two fields is specified.

**Review by artem-shelkovnikov — APPROVED** [review]

**Comment by artem-shelkovnikov** [comment]
> This should be removed from the template? https://github.com/orgs/elastic/teams/ingestion-team
I think so, good call!

File: lib/connectors_shared/elasticsearch_client.rb (hunk 1 above)
Comment (artem-shelkovnikov) [reply lib/connectors_shared/elasticsearch_client.rb:33]: Yep we can figure it out a bit later.

**Comment by wangch079** [comment]
>Do we also wanna allow for Basic Auth usage of Elasticsearch?

Although [it's officially supported](https://www.elastic.co/guide/en/elasticsearch/client/ruby-api/current/connecting.html#auth-basic), I am personally not a fan of Basic Auth. If I'm a customer, I'm not comfortable exposing my password in a yml file in clear text. 

But we can definitely add it if others in @elastic/ingestion-team feel it's necessary.

**Comment by artem-shelkovnikov** [comment]
Yeah totally agreed, I don't think it's necessary now tbh, just asking

**Comment by seanstory** [comment]
CC @vidok for awareness, as you'll probably want to build off of this PR for https://github.com/elastic/enterprise-search-team/issues/2108, so that you can use this Elasticsearch Client and not duplicate effort.

File: lib/connectors_shared/elasticsearch_client.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+      private
+
+      def client
+        @client ||= Elasticsearch::Client.new(connection_configs)
+      end
+
+      def connection_configs
+        es_config = ConnectorsApp::Config['elasticsearch']
+        configs = { :api_key => es_config['api_key'] }
```
[the remark this answers is not in the document]
Comment (wangch079) [reply lib/connectors_shared/elasticsearch_client.rb:28]: No, personal preference 😃

**Pull request merged**